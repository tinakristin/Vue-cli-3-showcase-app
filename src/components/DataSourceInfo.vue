<template>
  <div class="row pull-left">
    <font-awesome-icon icon="search" aria-hidden="true"/>
    <span v-if="hasDateInTranslation && fetchDate">
      {{ $t(translationPath + '.text', { dato: fetchDate }) }}
    </span>
    <span v-else>{{ $t(translationPath + '.text') }}</span>
    <a v-if=hasLinkInTranslation :href="$t(translationPath + '.link.url')"
       :target="hasTargetTranslation ? $t(translationPath + '.link.target') : '_blank'"
       :title="hasTitleTranslation ? $t(translationPath + '.link.title')
       : ('Følg lenken til ' + $t(translationPath + '.link.text'))">
      {{ $t(translationPath + '.link.text') }}
    </a>
  </div>
</template>

<script>
import _ from 'lodash';
import { library } from '@fortawesome/fontawesome-svg-core';
import { faSearch } from '@fortawesome/free-solid-svg-icons';

library.add(faSearch);

/**
   *
   * Lister ut info om datakilder.
   *
   * @props {translationPath, fetchDate, target, title}
   *
   * Properties:
   * 'translationPath': 'dataSourceInfo.bostedsInfo'
   * 'fetchDate': optional // Må oppgis dersom man forventer {dato} i translation-teksten
   *
   * Eksempel på forventet format på datakilder i translation-filen er:
   *
   * {
   *   text: 'Din bostedsadresse er hentet  den {dato} fra' // {dato} er optional
   *   link: { // Optional
   *       url: 'https://www.example.com',
   *       text: 'Example'
   *       target: optional // Hvis tom, default: '_blank'
   *       title: optional // Hvis tom, default: 'Følg lenken
 *         til'+$t(translationPath + '.link.text')
   *   }
   * }
   */
export default {
  uniqueId: _.uniqueId('dataopprinnelse_'),
  name: 'DataSourceInfo',
  props: {
    translationPath: {
      type: String,
      required: true,
    },
    fetchDate: {
      type: String,
      required: false,
    },
  },
  computed: {
    hasLinkInTranslation() {
      return _.includes(Object.keys(this.$t(`${this.translationPath}`)), 'link');
    },
    hasTargetTranslation() {
      return _.includes(Object.keys(this.$t(`${this.translationPath}.link`)), 'target');
    },
    hasTitleTranslation() {
      return _.includes(Object.keys(this.$t(`${this.translationPath}.link`)), 'title');
    },
    hasDateInTranslation() {
      return _.includes(this.$t(`${this.translationPath}.text`), '{dato}');
    },
  },
};
</script>

<style scoped lang="scss">
  .row {
    padding-left: 15px;
    font-size: 12px;
    color: #8699a4;
    a {
      color: #8699a4;
      text-decoration: underline;
    }
  }
</style>
