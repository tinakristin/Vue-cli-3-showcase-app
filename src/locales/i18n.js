import Vue from 'vue';
import VueI18n from 'vue-i18n';
import translation from './nb-NO/translation.json';

Vue.use(VueI18n);

export default new VueI18n({
  locale: 'no', // set locale
  fallbackLocale: 'no',
  messages: {
    no: translation, // set locale messages
  },
});
