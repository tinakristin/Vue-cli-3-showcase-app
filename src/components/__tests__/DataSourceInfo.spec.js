import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
import i18n from '../../locales/i18n';
import DataSourceInfo from '../DataSourceInfo.vue';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.component('font-awesome-icon', FontAwesomeIcon);

const propsData = {
  translationPath: 'dataSourceInfo.kartverket',
};

describe('DataSourceInfo', () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({ state: {} });
  });

  it('matches the snapshot', () => {
    const wrapper = shallowMount(DataSourceInfo, {
      localVue, i18n, store, propsData,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
