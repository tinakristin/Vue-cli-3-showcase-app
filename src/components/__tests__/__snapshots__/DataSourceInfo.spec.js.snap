// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSourceInfo matches the snapshot 1`] = `
<div class="row pull-left">
  <font-awesome-icon-stub icon="search" aria-hidden="true"></font-awesome-icon-stub> <span>Denne testdataen er hentet den {dato} fra </span> <a href="https://www.kartverket.no/data/brukerveiledning-for-adressesok/" target="_blank" title="Følg lenken til kartverkets addressesøk">
    kartverkets addressesøk
  </a>
</div>
`;
